### PR TITLE
drop reliance on old membership common models

### PR DESCRIPTION
**Pull request by johnduffell**
this PR incorporates https://github.com/guardian/membership-common/pull/331

removing the old model, and using the new back door to get the rate plan ids to remove discounts when people upgrade
cc @guardian/membership-and-subscriptions

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ object Dependencies {
   val sentryRavenLogback = "com.getsentry.raven" % "raven-logback" % "7.2.3"
   val scalaUri = "com.netaporter" %% "scala-uri" % "0.4.6"
   val memsubCommonPlayAuth = "com.gu" %% "memsub-common-play-auth" % "0.7"
-  val membershipCommon = "com.gu" %% "membership-common" % "0.285"
+  val membershipCommon = "com.gu" %% "membership-common" % "0.289-SNAPSHOT"
```
Comment (johnduffell) [inline project/Dependencies.scala]: todo - fill in this number once membership common is out :spiral_notepad:

File: frontend/app/services/MemberService.scala [hunk 2]
```diff
@@ -440,16 +440,20 @@ class MemberService(identityService: IdentityService,
     val newPlan = catalog.unsafeFindPaid(planChoice.productRatePlanId)
     val tier = newPlan.tier
 
-    val oldRest = zuoraService.getRestSubscription(sub.name).map(_.getOrElse(
-      throw new Exception(s"REST sub not found for ${sub.id}")
-    ))
+    List(sub.plan.id)
```
Comment (paulbrown1982) [inline frontend/app/services/MemberService.scala]: What is this for?

Comment (johnduffell) [inline frontend/app/services/MemberService.scala]: no idea, I'll delete!

**Comment by johnduffell** [comment]
@rtyley paul suggested you might want to take a look too if you have time?

**Pull request merged**